src/langcheck/metrics/_model_management.py

```diff
@@ -0,0 +1,219 @@
+from tabulate import tabulate
+from typing import Optional, Tuple, Union
+
+import pandas as pd
+import requests
+from configobj import ConfigObj
+from sentence_transformers import SentenceTransformer
+from transformers.models.auto.modeling_auto import \
+    AutoModelForSequenceClassification
+from transformers.models.auto.tokenization_auto import AutoTokenizer
+
+from ._model_loader import (load_auto_model_for_text_classification,
+                            load_sentence_transformers)
+
+# TODO: Use a ENUM class to parse these
+VALID_METRIC_NAME = [
+    'factual_consistency', 'toxicity', 'sentiment', 'semantic_similarity'
+]
+VALID_LANGUAGE = ['zh']
+VALID_LOADER = ['huggingface', 'sentence-transformers']
+
+
+class ModelManager:
+    '''
+    A class to manage different models for multiple languages in LangCheck.
+    This class allows setting and retrieving different model names (like
+    sentiment_model, semantic_similarity_model, etc.) for each language.
+    It also supports loading model configurations from a file.
+    '''
+
+    def __init__(self):
+        '''
+        Initializes the ModelConfig with empty model dictionaries for each
+        language.
+        '''
+        self.__init__config()
+        self.validate_config()
+
+    def __init__config(self):
+        cwd = os.path.dirname(__file__)
+        self.config = ConfigObj(
+            os.path.join(Path(cwd), 'config', 'metric_config.ini'))
+
+    @lru_cache
+    def fetch_model(
+        self, language: str, metric: str
+    ) -> Union[Tuple[AutoTokenizer, AutoModelForSequenceClassification],
+               SentenceTransformer]:
+        '''
+        Return the model used for the given metric and language.
+
+        Args:
+            language: The language for which to get the model
+            metric_type: The metric name
+        '''
+        if language in self.config:
+            if metric in self.config[language]:
+                # Deep copy the confguration so that changes to `config` would
+                # not affect the original `self.config`.
+                config = deepcopy(self.config[language][metric])
+                # Get model name, model loader type
+                model_name, loader_type = config['model_name'], config['loader']
+                # Check if the model version is fixed
+                revision = config.pop("revision", None)
+                if loader_type == 'sentence-transformers':
+                    if revision is not None:
+                        print(
+                            'Info: Sentence-Transformers do not support fixed model versions yet'  # NOQA:E501
+                        )
+                    model = load_sentence_transformers(model_name=model_name)
+                    return model
+                elif loader_type == 'huggingface':
+                    tokenizer_name = config.pop('tokenizer_name', None)
+                    tokenizer, model = load_auto_model_for_text_classification(
```

Comment:
For `factual_consistency`, the model is a text2text model so I think it needs to be downloaded with `AutoModelForSeq2SeqLM.from_pretrained` instead of `AutoModelForSequenceClassification.from_pretrained`. Maybe add another loader in `_model_loader.py`?